src/main/java/org/dbos/apiary/mysql/MysqlConnection.java

```diff
@@ -194,51 +197,7 @@ public boolean validate(Map<String, List<String>> writtenKeys, TransactionContex
             try {
                 c = commitConnection.get();
                 s = c.createStatement();
-            } catch (Exception e) {
-                e.printStackTrace();
-            }
-            while (true) {
-                try {
-                    s.clearBatch();
-                    for (String table : writtenKeys.keySet()) {
-                        String keyString = String.join(",", writtenKeys.get(table).stream().map(name -> ("'" + name + "'")).collect(Collectors.toList()));
-                        if (!keyString.isEmpty()) {
-                            query = String.format("UPDATE %s SET %s = %d WHERE %s IN (%s) AND %s < %d AND %s = %d", table, MysqlContext.endVersion, txc.txID, MysqlContext.apiaryID, keyString, MysqlContext.beginVersion, txc.txID, MysqlContext.endVersion, Long.MAX_VALUE);
-                            //s.execute(query);
-                            s.addBatch(query);
-                        }
-                    }
-                    s.executeBatch();
-                    if (delayLogFlush) {
-                        s.execute("FLUSH ENGINE LOGS;");
-                    }
-                } catch(BatchUpdateException e) {
-                    Throwable innerException = e.getCause();
-                    if (innerException instanceof MySQLTransactionRollbackException) {
-                        MySQLTransactionRollbackException m = (MySQLTransactionRollbackException)innerException;
-                        if (m.getErrorCode() == 1213 || m.getErrorCode() == 1205) {
-                            continue; // Deadlock or lock timed out
-                        }
-                    }
-                    e.printStackTrace();
-                    logger.error("1. Failed to update valid txn {}", txc.txID);
-                    logger.info("1. Validate update query: {}", query);
-                } catch (MySQLTransactionRollbackException m) {
-                    if (m.getErrorCode() == 1213 || m.getErrorCode() == 1205) {
-                        continue; // Deadlock or lock timed out
-                    } else {
-                        m.printStackTrace();
-                        logger.error("2. Failed to update valid txn {}", txc.txID);
-                        logger.info("2. Validate update query: {}", query);
-                    }
-                } catch (Exception e) {
-                    e.printStackTrace();
-                    logger.error("3. Failed to update valid txn {}", txc.txID);
-                    logger.info("3. Validate update query: {}", query);
-                }
-                break;
-            }
-            try {
+                s.execute("FLUSH ENGINE LOGS;");
```

Comment:
Do you want to put "FLUSH ENGINE LOGS" in an `if (delayLogFlush)` block? It would be easier to test with/without your optimization.